filebeat/harvester/source/source.go

```diff
@@ -0,0 +1,17 @@
+package source
+
+import (
+	"io"
+	"os"
+)
+
+type LogSource interface {
+	io.ReadCloser
+	Name() string
+}
+
+type FileSource interface {
+	LogSource
+	Stat() (os.FileInfo, error)
+	Continuable() bool // can we continue processing after EOF?
```

Comment:
I prefer not do logic changes when I do restructuring to make potential problems easier to track, so I would keep this for a future PR. To track all this stuff, I see two possibilities: Meta Github issue or TODO in the code. What do you prefer? Github issue is probably better as it is easier to track progress.

I checked Stat() and Continuable() and it seems like both are still used in the reader.

I get the feeling as soon as we fully restructured harvester the above stuff will cleanup up by itself.